tests/test_json_schema.py

```diff
@@ -3334,139 +3334,143 @@ class Model(BaseModel):
+                            'discriminator': {
+                                'propertyName': 'color',
+                                'mapping': {
+                                    'PetColor.black': {
+                                        'oneOf': [
+                                            {'$ref': '#/$defs/BlackCatWithHeight'},
+                                            {'$ref': '#/$defs/BlackCatWithWeight'},
+                                        ],
+                                        'discriminator': {
+                                            'propertyName': 'info',
+                                            'mapping': {
+                                                'PetInfo.height': '#/$defs/BlackCatWithHeight',
+                                                'PetInfo.weight': '#/$defs/BlackCatWithWeight',
+                                            },
                                         },
-                                        'propertyName': 'info',
                                     },
-                                    'oneOf': [
-                                        {'$ref': '#/$defs/BlackCatWithHeight'},
-                                        {'$ref': '#/$defs/BlackCatWithWeight'},
-                                    ],
+                                    'PetColor.white': '#/$defs/WhiteCat',
                                 },
-                                'white': '#/$defs/WhiteCat',
                             },
-                            'propertyName': 'color',
                         },
-                        'oneOf': [
-                            {
-                                'discriminator': {
-                                    'mapping': {'0': '#/$defs/BlackCatWithHeight', '1': '#/$defs/BlackCatWithWeight'},
-                                    'propertyName': 'info',
-                                },
-                                'oneOf': [
-                                    {'$ref': '#/$defs/BlackCatWithHeight'},
-                                    {'$ref': '#/$defs/BlackCatWithWeight'},
-                                ],
-                            },
-                            {'$ref': '#/$defs/WhiteCat'},
-                        ],
+                        'PetType.dog': '#/$defs/Dog',
                     },
-                    {'$ref': '#/$defs/Dog'},
-                ],
+                },
                 'title': 'Pet',
             },
+            'number': {'type': 'integer', 'title': 'Number'},
         },
         'required': ['pet', 'number'],
         'title': 'Model',
-        'type': 'object',
+        '$defs': {
+            'BlackCatWithHeight': {
+                'type': 'object',
+                'properties': {
+                    'pet_type': {'const': 'cat', 'title': 'Pet Type'},
+                    'color': {'const': 'black', 'title': 'Color'},
+                    'info': {'const': 0, 'title': 'Info'},
```

Comment:
however we are getting `0` or `'cat'` or `'black'` here is how we should be populating the keys of the discriminator mapping above